sdk/formrecognizer/Azure.AI.FormRecognizer/samples/Sample4_StronglyTypingARecognizedForm.md

```diff
@@ -0,0 +1,108 @@
+# Strongly-typing a recognized form
+
+This sample demonstrates how to use the fields in your recognized forms to create an object with strongly-typed fields. The output from the `StartRecognizeReceipts` method will be used to illustrate this sample, but note that a similar approach can be used for any custom form as long as you properly update the fields' names and types.
+
+To get started you'll need a Cognitive Services resource or a Form Recognizer resource.  See [README][README] for prerequisites and instructions.
+
+## Creating a `FormRecognizerClient`
+
+To create a new `FormRecognizerClient` you need the endpoint and credentials from your resource. In the sample below you'll use a Form Recognizer API key credential by creating an `AzureKeyCredential` object, that if needed, will allow you to update the API key without creating a new client.
+
+You can set `endpoint` and `apiKey` based on an environment variable, a configuration setting, or any way that works for your application.
+
+```C# Snippet:CreateFormRecognizerClient
+string endpoint = "<endpoint>";
+string apiKey = "<apiKey>";
+var credential = new AzureKeyCredential(apiKey);
+var client = new FormRecognizerClient(new Uri(endpoint), credential);
+```
+
+## Creating a wrapper class
+
+Recognized receipts and custom forms are returned as `RecognizedForm` objects from the Form Recognizer client. Even though all recognized information can be obtained from them, sometimes they can be inconvenient to handle and, for this reason, we'll illustrate how to create a wrapper class to make the relevant fields easily accessible.
+
+We'll be using a recognized receipt as a sample, so we'll call our wrapper class `Receipt`. To store items listed in the receipt, we'll use a similar wrapper called `ReceiptItem`. To see the full source files, see:
+
+* [Receipt class](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/formrecognizer/Azure.AI.FormRecognizer/tests/samples/Receipt.cs)
+* [ReceiptItem class](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/formrecognizer/Azure.AI.FormRecognizer/tests/samples/ReceiptItem.cs)
+
+The `Receipt` class is composed of multiple `FormField<T>` properties. `FormField<T>` is a class defined in the main Form Recognizer library and used as a strongly-typed version of `FormField`, and it's more convenient to handle since there's no need to perform type checking.
+
+`Receipt` has methods for converting a `FormField` into a strongly-typed `FormField<T>`. These methods are used in the constructor to convert fields we expect the service to return for a receipt, such as `MerchantName` or `TransactionDate`. You can use the same methods when writing a wrapper class for your custom forms, but you need to update the fields' names and types accordingly.
+
+```C# Snippet:FormRecognizerSampleReceiptWrapper
+public Receipt(RecognizedForm recognizedForm)
```

Comment:
In java, we are keeping the `ReceiptExtensions` class that allows the users to directly use these conversions with `ReceiptExtensions.asUSReceipt(recognizedForm)` . Any reason we don't include it in .NET?

Reply:
Yes. We are not trying to extend the Receipt functionality in this sample. We're just showing customers a good way of parsing recognized forms, using Receipt as an example. `ReceiptExtensions.asUSReceipt` is not necessarily applicable to other forms, and we're not even sure if that's the path we want to follow for receipts anyway (which made us give up on the strong `RecognizedReceipt` type).